src/NpcAdventureModApi.cs

```diff
@@ -0,0 +1,72 @@
+﻿using NpcAdventure.StateMachine;
+using NpcAdventure.StateMachine.State;
+using StardewValley;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+
+namespace NpcAdventure
+{
+    public class NpcAdventureModApi
+    {
+        private NpcAdventureMod npcAdventureMod;
+
+        internal NpcAdventureModApi(NpcAdventureMod npcAdventureMod)
+        {
+            this.npcAdventureMod = npcAdventureMod;
+        }
+        public bool CanRecruit()
+        {
+            return npcAdventureMod != null && npcAdventureMod.CompanionManager != null && npcAdventureMod.CompanionManager.CanRecruit();
+        }
+        public string[] GetPossibleCompanions()
+        {
+            return npcAdventureMod.CompanionManager.PossibleCompanions.Keys.ToArray();
+        }
+        public bool IsPossibleCompanion(string npc)
+        {
+            return npcAdventureMod != null && npcAdventureMod.CompanionManager != null && npcAdventureMod.CompanionManager.PossibleCompanions != null && npcAdventureMod.CompanionManager.PossibleCompanions.ContainsKey(npc);
```

Comment:
This line is too long. I recomend to add some break lines for better readability, or use question marks for implicit nullchecks:
```suggestion
            return npcAdventureMod?.CompanionManager?.PossibleCompanions.ContainsKey(npc);
```

Reply:
the null checks make it return a nullable bool instead... is there a way to avoid that?

Reply:
````cs
bool IsSomeOkay()
{
    return MyInstance.someNullable?.boolField == true;
}
```

Reply:
oh, right! ty